### PR TITLE
in_opentelemetry: attempt to fix tag_from_uri

### LINKED ISSUE #8734: tag_from_uri in opentelemetry input plugin has no effect
## Bug Report

**Describe the bug**
if configuring `tag_from_uri` on the opentelemetry input plugin, tags are not updated

**To Reproduce**
- Steps to reproduce the problem:
given this config:
```
pipeline:
    inputs:
        - name: opentelemetry
          listen: 0.0.0.0
          port: 4320
          tag_from_uri: "true"
    outputs:
        - name: 'stdout'
          match: '*'
```

And this curl
```
curl --header "Content-Type: application/json" --request POST --data '{"resourceLogs":[{"resource":{},"scopeLogs":[{"scope":{},"logRecords":[{"timeUnixNano":"1660296023390371588","body":{"stringValue":"{\"message\":\"dummy\"}"},"traceId":"","spanId":""}]}]}]}'   http://localhost:4320/v1/logs
```
I would expect the message to be tagged with `v1_logs`. However, it's not retagged, the tag is still `opentelemetry.0`

**Expected behavior**
tags should be changed to reflect the uri

**Your Environment**

* Version used: current version in `cr.fluentbit.io/fluent/fluent-bit`
* Configuration: see above
* Environment name and version (e.g. Kubernetes? What version?): regular docker

**Additional context**
without this functionality it is very difficult to route opentelemetry data (for example logs to loki, metrics to prometheus etc) using fluent-bit.

### PR DESCRIPTION
**Pull request by shaohme**
Pass on tag with length so that tag_from_uri get applied on OpenTelemetry input plugin

Fixes #8734 
----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [x] Example configuration file for the change
```
service:
  flush: 1
  log_file: /tmp/fluent-bit.log
  daemon: false
  log_level: debug
pipeline:
  inputs:
    - name: opentelemetry
      listen: 127.0.0.1
      port: 9999
      raw_traces: false
      successful_response_code: 200
      tag_from_uri: true
      http2: false
  outputs:
    - name: stdout
      match: "v1_traces"
```
- [x] Debug log output from testing the change
```
[2024/05/27 14:58:22] [ info] [fluent bit] version=3.0.7, commit=dffcd42ada, pid=1820178
[2024/05/27 14:58:22] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2024/05/27 14:58:22] [ info] [storage] ver=1.5.2, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2024/05/27 14:58:22] [ info] [cmetrics] version=0.9.0
[2024/05/27 14:58:22] [ info] [ctraces ] version=0.5.1
[2024/05/27 14:58:22] [ info] [input:opentelemetry:opentelemetry.0] initializing
[2024/05/27 14:58:22] [ info] [input:opentelemetry:opentelemetry.0] storage_strategy='memory' (memory only)
[2024/05/27 14:58:22] [debug] [opentelemetry:opentelemetry.0] created event channels: read=22 write=18
[2024/05/27 14:58:22] [debug] [downstream] listening on 127.0.0.1:9999
[2024/05/27 14:58:22] [ info] [input:opentelemetry:opentelemetry.0] listening on 127.0.0.1:9999
[2024/05/27 14:58:22] [debug] [stdout:stdout.0] created event channels: read=24 write=25
[2024/05/27 14:58:22] [ info] [sp] stream processor started
[2024/05/27 14:58:22] [ info] [output:stdout:stdout.0] worker #0 started
[2024/05/27 14:58:24] [debug] [task] created task=0x7f2811436640 id=0 OK
[2024/05/27 14:58:24] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[2024/05/27 14:58:24] [debug] [output:stdout:stdout.0] ctr decode msgpack returned : 6
[2024/05/27 14:58:24] [debug] [out flush] cb_destroy coro_id=0
[2024/05/27 14:58:24] [debug] [task] destroy task=0x7f2811436640 (task_id=0)
[2024/05/27 14:58:25] [debug] [task] created task=0x7f28114366e0 id=0 without routes, dropping.
[2024/05/27 14:58:25] [debug] [task] destroy task=0x7f28114366e0 (task_id=0)
[2024/05/27 14:58:25] [debug] [task] created task=0x7f28114366e0 id=0 OK
[2024/05/27 14:58:25] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[2024/05/27 14:58:25] [debug] [output:stdout:stdout.0] ctr decode msgpack returned : 6
[2024/05/27 14:58:25] [debug] [out flush] cb_destroy coro_id=1
[2024/05/27 14:58:25] [debug] [task] destroy task=0x7f28114366e0 (task_id=0)
[2024/05/27 14:58:26] [debug] [task] created task=0x7f28114365a0 id=0 without routes, dropping.
[2024/05/27 14:58:26] [debug] [task] destroy task=0x7f28114365a0 (task_id=0)
[2024/05/27 14:58:26] [debug] [task] created task=0x7f28114365a0 id=0 OK
[2024/05/27 14:58:26] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[2024/05/27 14:58:26] [debug] [output:stdout:stdout.0] ctr decode msgpack returned : 6
[2024/05/27 14:58:26] [debug] [out flush] cb_destroy coro_id=2
[2024/05/27 14:58:26] [debug] [task] destroy task=0x7f28114365a0 (task_id=0)
[2024/05/27 14:58:28] [ warn] [engine] service will shutdown in max 5 seconds
[2024/05/27 14:58:28] [ info] [engine] service has stopped (0 pending tasks)
[2024/05/27 14:58:28] [ info] [output:stdout:stdout.0] thread worker #0 stopping...
[2024/05/27 14:58:28] [ info] [output:stdout:stdout.0] thread worker #0 stopped
```
- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [N/A] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by nuclearpidgeon** [comment]
I'm at a loss as to why this was never implemented in the first place... perhaps at some point there was an issue with tag-based routing at the point this code was written ¯\_(ツ)_/¯. Nice work fixing it up! This was what I was hoping to PR at some point myself.

**Review by nuclearpidgeon — APPROVED** [review]

**Comment by cosmo0920** [comment]
Hi, fluent organization requires to add DCO. So, could you add Signed-off in your commit?
The description is here: https://github.com/fluent/fluent-bit/pull/8881/checks?check_run_id=25458914982

**Comment by shaohme** [comment]
> Hi, fluent organization requires to add DCO. So, could you add Signed-off in your commit? The description is here: https://github.com/fluent/fluent-bit/pull/8881/checks?check_run_id=25458914982

Sure. Should be added now. Had to force push.

**Comment by cosmo0920** [comment]
Plus, we need to add prefix for changed module name into commit message(s).
In this case, we need to use "in_opentelemery: attempt to fix tag_from_uri" in your commit message.

**Comment by shaohme** [comment]
> Plus, we need to add prefix for changed module name into commit message(s). In this case, we need to use "in_opentelemery: attempt to fix tag_from_uri" in your commit message.
> 
> see also: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#commit-changes

Should be fixed now. And I should probably have read a contributors manual first ... :smirk:

File: plugins/in_opentelemetry/opentelemetry_prot.c [hunk 1]
```diff
@@ -1736,6 +1741,8 @@ int opentelemetry_prot_handle(struct flb_opentelemetry *ctx, struct http_conn *c
         }
     }
 
+    size_t tag_len = flb_sds_len(tag);
```
Comment (cosmo0920) [inline plugins/in_opentelemetry/opentelemetry_prot.c:1744]: Declaring middle of the function is not good coding style on Fluent Bit.
Could you declare a variable in the beginning of this function? And then, calculate length of tag here should be fine.

Comment (shaohme) [reply plugins/in_opentelemetry/opentelemetry_prot.c]: Sure can. Should be pushed now.

Comment (shaohme) [reply plugins/in_opentelemetry/opentelemetry_prot.c]: > Declaring middle of the function is not good coding style on Fluent Bit. Could you declare a variable in the beginning of this function? And then, calculate length of tag here should be fine.

Sure can. Should be pushed now.

**Review by cosmo0920 — APPROVED** [review]
Looks fine to me.

**Pull request merged**